### PR TITLE
Fix/login errors not displaying

### LINKED ISSUE #225: Refactor internalFetch to handle api errors nicely
@sindelio Please fill it.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Jakedavies**
closes #225 #200 

I probably need to rebase and do a little bit of refactoring against master so the new internalFetch works for everything, but I think overall this nets us a lot less duplicated code

### CODE REVIEW

**Comment by Jakedavies** [comment]
This also allows internalFetch to be used when the user isn't login (checks for token existence before attaching it)

**Comment by Jakedavies** [comment]
An important thing to note here is that we parse the json body by default now with internalFetch, to save everyone parsing it themselves. So when consuming internalFetch you don't need to await res.json()

File: admin/src/containers/Users/ChangePassword.js [hunk 1]
```diff
@@ -18,44 +18,18 @@ export class ChangePasswordContainer extends React.Component {
   };
 
   async changePassword(userId, password) {
-    const response = await internalFetch(
```
Comment (maksymilian-majer) [inline admin/src/containers/Users/ChangePassword.js]: Is it supposed to affect this file?
I think there's some merge gone wrong.

File: admin/src/containers/Users/Users.js [hunk 2]
```diff
@@ -19,9 +19,7 @@ export class UsersContainer extends React.Component {
   };
 
   async getUsers() {
-    const response = await internalFetch(`${config.API_URL}/api/management/users/list`);
-
-    return await response.json();
+    return await internalFetch(`${config.API_URL}/api/management/users/list`);
```
Comment (maksymilian-majer) [inline admin/src/containers/Users/Users.js]: Same here. I think the new master is already using the context here.

**Pull request closed without merging**